src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -225,5 +233,76 @@ public override void OnEventWritten(string name, object? payload)
                 break;
         }
     }
+
+    private void RecordDuration(Activity? activity, object? payload, bool hasError = false)
+    {
+        TagList tags = default(TagList);
+
+        if (activity != null && activity.IsAllDataRequested)
+        {
+            var names = new[]
```

Comment:
This can be a static list to avoid allocating an array every time a measurement is recorded.